File: WebApplication1/Models/Student.cs
using Models;
using System;
using System.ComponentModel.DataAnnotations;

namespace WebApplication1.Models
{
    public class Student
    {
        public int StudentId { get; set; }
        public int UserId { get; set; }  // Foreign Key
        [Required]
        [MaxLength(50)]
        public string FirstName { get; set; }

        [Required]
        [MaxLength(50)]
        public string LastName { get; set; }

        [Required]
        [MaxLength(100)]
        [EmailAddress]
        public string Email { get; set; }
        public string Phone { get; set; }
        public string Instrument { get; set; }
        public string HighSchool { get; set; }
        public string ProfilePicture { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;


        // Navigation Properties
        public User User { get; set; }
        public ICollection<Video> Videos { get; set; }
        public ICollection<Offer> Offers { get; set; }
        public ICollection<InterestedStudent> InterestedStudents { get; set; }
    }
}

File: client/src/models/student.ts
import { User } from './User';
import { Video } from './Video';
import Offer  from './Offer';
import Interest  from './Interest';
import Rating from './Rating';

export default interface Student extends User {
    firstName: string;
    lastName: string;
    phone: string;
    graduationYear: number;
    instrument: string;
    highSchool: string;
    profilePicture?: string; // Optional
    createdAt: string;
    averageRating?: number;
    videos?: Video[]; // List of videos uploaded by the student
    scholarshipOffers?: Offer[]; // List of offers received by the student
    interests?: Interest[]; 
    rating?: Rating[];// List of interests expressed by the student
}
